src/lib/mina_block/block.mli

```diff
@@ -5,28 +5,28 @@ open Mina_transaction
 module Stable : sig
   [@@@no_toplevel_latest_type]
 
-  module V1 : sig
-    type t [@@deriving compare, sexp, to_yojson]
+  module V2 : sig
+    type t [@@deriving sexp, to_yojson, equal]
   end
 end]
 
-type t = Stable.Latest.t [@@deriving compare, sexp, to_yojson]
+type t = Stable.Latest.t [@@deriving sexp, to_yojson, equal]
 
 type with_hash = t State_hash.With_state_hashes.t [@@deriving sexp]
 
 (* TODO: interface for both unchecked and checked construction of blocks *)
 (* check version needs to run following checks:
      - Header.verify (could be separated into header construction)
-     - Body_reference.verify_reference header.body_reference body
-     - Body.verify (cannot be put into body construction as we should do the reference check first, but could be separated) *)
+     - Consensus.Body_reference.verify_reference header.body_reference body
```

Comment:
These choices of location seem odd. Is there a reason that they can't live in this library instead?